core/lib/Cardano/Address/Script.hs

```diff
@@ -801,9 +801,11 @@ instance FromJSON ScriptTemplate where
         ScriptTemplate <$> (Map.fromList <$> cosigners') <*> template'
       where
         parseCosignerPairs = withObject "Cosigner pairs" $ \o ->
-            case JM.toList o of
+            case toList o of
                 [] -> fail "Cosigners object array should not be empty"
                 cs -> for (reverse cs) $ \(numTxt, str) -> do
                     cosigner' <- parseJSON @Cosigner (String (J.keyToText numTxt))
                     xpub <- parseXPub str
                     pure (cosigner', xpub)
+
+        toList = JM.foldlWithKey' (\acc k v -> (k,v):acc) []
```

Comment:
I think we could remove the imports
```
import qualified HaskellWorks.Data.Aeson.Compat as J
import qualified HaskellWorks.Data.Aeson.Compat.Map as JM
```
and use https://hackage.haskell.org/package/aeson-2.1.0.0/docs/Data-Aeson-KeyMap.html
(KeyMap is Foldable)

Reply:
possibly we don't need hw-aeson at all, and use functionality exposed from `Data.Aeson.KeyMap`

Reply:
`Data.Aeson.Keymap` isn't exported in aeson versions < 2.0.0.0, which is why I think we're using `hw-aeson`, as a compatibility shim.

Reply:
And not all our users have migrated to Aeson 2 yet unfortunately.

Reply:
aha, I got it. Thanks for the explanation!